pkg/templates/livenessport/template.go

```diff
@@ -0,0 +1,57 @@
+package livenessport
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/livenessport/internal/params"
+	"golang.stackrox.io/kube-linter/pkg/templates/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const templateKey = "liveness-http-port"
+
+var sentinal = struct{}{}
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Liveness Port Not Open",
+		Key:         templateKey,
+		Description: "Flag containers have a http liveness prop with a port they didn't open.",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return util.PerNonInitContainerCheck(func(container *v1.Container) []diagnostic.Diagnostic {
+				if container.LivenessProbe == nil {
+					return nil
+				}
+
+				httpProbe := container.LivenessProbe.ProbeHandler.HTTPGet
```

Comment:
How about extending this check for TCP ?

Reply:
Good idea adding a check since UDP sockets would fail. 

```
if port.Protocol != "" && port.Protocol != "TCP" {
    continue
}
```